src/renderer/atlas/AtlasEngine.api.cpp

```diff
@@ -637,28 +670,30 @@ void AtlasEngine::_resolveFontMetrics(const wchar_t* requestedFaceName, const Fo
 
     if (fontMetrics)
     {
-        const auto underlineOffsetInPx = static_cast<double>(-metrics.underlinePosition) * designUnitsPerPx;
-        const auto underlineThicknessInPx = static_cast<double>(metrics.underlineThickness) * designUnitsPerPx;
-        const auto strikethroughOffsetInPx = static_cast<double>(-metrics.strikethroughPosition) * designUnitsPerPx;
-        const auto strikethroughThicknessInPx = static_cast<double>(metrics.strikethroughThickness) * designUnitsPerPx;
-        const auto lineThickness = gsl::narrow<u16>(std::round(std::min(underlineThicknessInPx, strikethroughThicknessInPx)));
-        const auto underlinePos = gsl::narrow<u16>(std::ceil(baseline + underlineOffsetInPx - lineThickness / 2.0));
-        const auto strikethroughPos = gsl::narrow<u16>(std::round(baseline + strikethroughOffsetInPx - lineThickness / 2.0));
-
-        auto fontName = wil::make_process_heap_string(requestedFaceName);
-        const auto fontWeight = gsl::narrow<u16>(requestedWeight);
+        std::wstring fontName{ requestedFaceName };
+        const auto fontWeightU16 = gsl::narrow<u16>(requestedWeight);
+        const auto underlinePosU16 = gsl::narrow<u16>(underlinePos);
+        const auto underlineWidthU16 = gsl::narrow<u16>(underlineWidth);
+        const auto strikethroughPosU16 = gsl::narrow<u16>(strikethroughPos);
+        const auto strikethroughWidthU16 = gsl::narrow<u16>(strikethroughWidth);
+        const auto doubleUnderlinePosTopU16 = gsl::narrow<u16>(doubleUnderlinePosTop);
+        const auto doubleUnderlinePosBottomU16 = gsl::narrow<u16>(doubleUnderlinePosBottom);
+        const auto thinLineWidthU16 = gsl::narrow<u16>(thinLineWidth);

```

Comment:
all throwing -- OK?

Reply:
Technically I only want to throw if the result is out of bounds for an `uint16_t`. This code _should_ work because `float`s can represent 16-bit integers exactly (i.e. converting back and forth yields the same result). I could convert them to `int32_t` first and then do the narrowing, e.g. with `lround`. What do you think?

Reply:
Whatever you think strikes the best clarity vs work balance. This is fine with me :)

Reply:
Is this perhaps being unnecessarily cautious? I means is there any scenario here where a simple `narrow_cast` would actually produce a worse outcome than a throw?

Reply:
_Personally_, I tend to prefer `gsl::narrow` over `gsl::narrow_cast` unless I need to the latter for performance or exception safety (or I just want narrowing). I don't think the cost of the narrowing check is particularly high in most cases, so I'm fine with it.

In this case I'm a bit afraid that these values could be negative or larger than `UINT16_MAX` (since they come from the font file which might specify anything). I'll probably modify this PR to use `gsl::narrow<u16>(lroundf(...))` which even more excessively cautious, but it avoids any potential rounding errors, while catching any weird behavior (e.g. font size of >9000 leads to cell size >65536) and is fast enough (this code is only called once and each `lroundf` is <10ns).

Reply:
What I was getting at is say you have some kind of overflow, and your underline position is now so large that it's off screen, is that a big deal? Is it likely to crash? If not, and the alternative is that the font just doesn't work at all, it doesn't seem like the `narrow_cast` is really an improvement. But it's not important. It was just a passing thought.